Tasks/DotNetCoreCLIV2/dotnetcore.ts

```diff
@@ -325,22 +326,50 @@ export class dotNetExe {
         }
 
         var projectFiles = utility.getProjectFiles(projectPattern);
+        var resolvedProjectFiles: string[] = [];
 
         if (searchWebProjects) {
-            projectFiles = projectFiles.filter(function (file, index, files): boolean {
+            resolvedProjectFiles = projectFiles.filter(function (file, index, files): boolean {
                 var directory = path.dirname(file);
                 return tl.exist(path.join(directory, "web.config"))
                     || tl.exist(path.join(directory, "wwwroot"));
             });
 
-            if (!projectFiles.length) {
-                tl.error(tl.loc("noWebProjectFound"));
-            }
+            if (!resolvedProjectFiles.length) {
+                var projectFilesUsingWebSdk = projectFiles.filter(this.isWebSdkUsed)
+                if(!projectFilesUsingWebSdk.length) {
+                    tl.error(tl.loc("noWebProjectFound"));
+                }
+                return projectFilesUsingWebSdk;
+            } 
+            return resolvedProjectFiles;
         }
-
         return projectFiles;
     }
 
+    private isWebSdkUsed(projectfile: string): boolean {
+        try {
+            var fileBuffer: Buffer = fs.readFileSync(projectfile);
+            var webConfigContent: string = fileBuffer.toString();
+            
+            var fileEncodings = ['utf8', 'utf16le', 'ascii', 'base64', 'binary', 'hex']
+
+            for(var i = 0; i < fileEncodings.length; i++) {
+                tl.debug("Trying to decode with " + fileEncodings[i])
+                webConfigContent = fileBuffer.toString(fileEncodings[i])
+                try {
+                    var projectSdkUsed: string = ltx.parse(webConfigContent).getAttr("sdk") || ltx.parse(webConfigContent).getAttr("Sdk")
+                    if(projectSdkUsed != undefined && projectSdkUsed.toLowerCase() == "microsoft.net.sdk.web") return true
+                } catch (error) {}
```

Comment:
should we go over the for loop only in case of error. You can achieve this by "return projectSdkUsed && projectSdkUsed.toLowerCase() == "microsoft.net.sdk.web""

Reply:
Currently, we don't iterate the loop if we found Sdk.. removing if condition would let the loop run it's first iteration only..

Reply:
yes lets say we found the sdk tag but it is not web then we should not iterate

Reply:
Done